creator-node/src/routes/files.js

```diff
@@ -174,69 +181,69 @@ module.exports = function (app) {
     logger.info(`IPFS Standalone Request - ${CID}`)
     logger.info(`IPFS Stats - Standalone Requests: ${totalStandaloneIpfsReqs}`)
 
-    // Conditionally rehydrate from filestorage to IPFS.
-    try {
-      await rehydrateIpfsFromFsIfNecessary(
-        req,
-        CID,
-        queryResults.storagePath
-      )
-    } catch (e) {
-      // If rehydrate throws error, return 500 without attempting to stream file.
-      return sendResponse(req, res, errorResponseServerError(e.message))
-    }
-
     // If client has provided filename, set filename in header to be auto-populated in download prompt.
     if (req.query.filename) {
       res.setHeader('Content-Disposition', contentDisposition(req.query.filename))
     }
 
-    // Stream file to client.
-    try {
-      // Cat 1 byte of CID in ipfs to determine if file exists
-      // If the request takes under 500ms, stream the file from ipfs
-      // else if the request takes over 500ms, throw an error and stream the file from file system
-      await ipfsSingleByteCat(CID, req, 500)
+    if (req.query.fromFS && req.query.fromFS === 'true') {
+      logger.info(`Retrieving directly from filesystem`)
+      // Retrieves the file directly from the filesystem rather than checking IPFS first
+
+      // Fire-and-forget rehydration
+      rehydrateIpfsFromFsIfNecessary(
+        req,
+        CID,
+        queryResults.storagePath
+      )
 
-      // Stream file from ipfs if cat one byte takes under 500ms
-      // If catReadableStream() promise is rejected, throw an error and stream from file system
-      await new Promise((resolve, reject) => {
-        req.app.get('ipfsAPI').catReadableStream(CID)
-          .on('data', streamData => { res.write(streamData) })
-          .on('end', () => { res.end(); resolve() })
-          .on('error', e => { reject(e) })
-      })
-    } catch (e) {
-      // ipfsCatSingleByte took over 500ms, try streaming from file system and
-      // return the response from helper method
       return streamFromFileSystem(req, res, queryResults.storagePath)
-    }
-  })
+    } else {
+      logger.info(`Attempting to retrieve from IPFS`)
+      // Retrieves the file from IPFS and falls back to the filesystem if unavailable
 
-  // Helper method to stream file from file system on creator node
-  const streamFromFileSystem = async (req, res, path) => {
-    try {
-      // If file cannot be found on disk, throw error
-      if (!fs.existsSync(path)) {
-        return sendResponse(req, res, errorResponseServerError('File could not be found on disk.'))
+      // Conditionally rehydrate from filestorage to IPFS.
+      try {
+        await rehydrateIpfsFromFsIfNecessary(
+          req,
+          CID,
+          queryResults.storagePath
+        )
+      } catch (e) {
+        // If rehydrate throws error, return 500 without attempting to stream file.
+        return sendResponse(req, res, errorResponseServerError(e.message))
```

Comment:
if rehydration fails, shouldn't we still try to stream the song from fs? / why not attempt to try

Reply:
this is a good Q, but also what we were doing before (i didn't change this aspect of it. maybe we do want to though)

Reply:
hmm yeah, @dmanjunath was there a reason why we did this? looks like most of the errors are caught in rehydrateIpfsFromFsIfNecessary() anyway so i dont think it'll really hit the catch block here